working/0723-static-extensions/feature-specification.md

```diff
@@ -315,165 +453,482 @@ extension.
 
 #### Declarations of constructors in extensions
 
-This proposal relies on the [generic constructor proposal][]. In
-particular, this proposal uses concepts and definitions from the generic
-constructor proposal, and it is assumed that generic constructors are
-supported by the underlying Dart language.
+With this proposal, it is now supported to declare a constructor in an
+extension, with the same syntax as in a class and in other type
+introducing membered declarations.
 
-[generic constructor proposal]: https://github.com/dart-lang/language/pull/4265
+*The semantic interpretation of constructors declared in extensions
+follows closely from the informal interpretation given above in which
+the constructor is viewed as a static method whose type parameters are
+the type parameters of the extension in which it is declared
+(including their bounds) and whose return type is the on-type of the
+extension (which may contain references to said type parameters).*
 
-With this proposal, it is also supported to declare a generic constructor
-in an extension, with the same syntax as in a class and in other type
-introducing membered declarations.
+It is a compile-time error to declare a constructor with no type
+parameters in an extension whose on-type is not regular-bounded,
+assuming that the type parameters declared by the extension satisfy
+their bounds.
+
+*There is nothing semantically problematic with such a constructor.
+The semantic generic method to which it corresponds is well-defined
+and statically valid.  However, constructors on classes can be assumed
+to produce regular-bounded types, and so it seems reasonable to impose
+the same discipline on constructors added via extensions.*
 
 It is a compile-time error if an extension declares a generic constructor
```

Comment:
Constructors are also not generic.
(The corresponding static function might be.)